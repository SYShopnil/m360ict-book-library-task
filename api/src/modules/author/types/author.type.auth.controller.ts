export interface IAuthorLoginBOdy {
  password: string;
  email: string;
}
export interface IPayloadDataOfAccessToken {
  id: number;
  email: string;
}
