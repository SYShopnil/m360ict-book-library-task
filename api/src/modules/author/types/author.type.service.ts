import { IAuthor } from '../../../type/entity';

export interface IGetAllAuthors {
  authors: IAuthor[];
  total: number;
  totalPages: number;
}

export interface IRGetAuthorById extends IAuthor {
  email: string;
}
export interface IRGetAuthorByEmail extends IAuthor {
  email: string;
  password: string;
}
