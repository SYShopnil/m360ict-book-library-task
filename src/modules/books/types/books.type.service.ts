import { IBook } from '../../../type/entity';

export interface GetBooksOptions {
  page: number;
  pageSize: number;
  title?: string;
  author?: string;
}

export interface IRGetBookById extends IBook {}
export interface IRGetBooksByAuthor extends IBook {
  author_name: string;
  authors_bio: string;
}
