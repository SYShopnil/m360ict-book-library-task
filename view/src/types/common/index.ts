export interface ICommonReturnData {
  message: string;
  status: number;
}

export enum EAuth {
  AuthTokenCookieName = "auth",
}

export enum EDataTestId {
  cLoginFormWithSubmit = "cLoginFormWithSubmit",
  cLogoutContainer = "cLogoutContainer",
  rUserList = "rUserList",
  SHeaderMain = "SHeaderMain",
  SProductSection = "SProductSection",
  SProfile = "SProfile",
  CPaginationTrack = "CPaginationTrack",
  RSingleProduct = "RSingleProduct",
  RTableBody = "RTableBody",
  SProductCard = "SProductCard",
  CProductSearchBarContainer = "CProductSearchBarContainer",
}
