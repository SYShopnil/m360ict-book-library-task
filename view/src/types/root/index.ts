export type { IIconStore } from "./_icon";
export { IconName as IconEnums } from "./_icon";
export type { IButton, INormalButton, IRedirectButton } from "./Button";
export { BtnColorSchema } from "./Button";
