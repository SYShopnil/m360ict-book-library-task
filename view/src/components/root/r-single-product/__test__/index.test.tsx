import { render, screen } from "@testing-library/react";
import { RSingleProduct } from "..";
import { EDataTestId } from "@src/types/common";

describe("When Single Product has rendered", () => {
  it("Expect render successfully", async () => {
    // const { getByTestId } = render(
    //   await RSingleProduct({
    //     requestFetchForSingleProduct: new Promise((resolve) =>
    //       resolve({
    //         message: "Product Found!!!",
    //         payload: {
    //           product: {
    //             productId: "1",
    //             image: "/assert/demo-product.jpg",
    //             name: "Computer",
    //             price: "50000",
    //             desc: "Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. ",
    //             category: "Technology",
    //           },
    //         },
    //         status: 202,
    //       })
    //     ),
    //   })
    // );
    // const myElement = getByTestId(EDataTestId.RSingleProduct);
    const myElement = screen.getByTestId(EDataTestId.RSingleProduct);
    expect(myElement).toMatchSnapshot();
  });
});
