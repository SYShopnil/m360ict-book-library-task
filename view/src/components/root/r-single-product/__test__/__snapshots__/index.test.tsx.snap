// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When Single Product has rendered Expect render successfully 1`] = `
<div
  class="grid grid-cols-12 gap-2 px-[1.5rem]"
  data-testid="RSingleProduct"
>
  <div
    class="col-span-12 lg:col-span-5 flex justify-center items-center"
  >
    <img
      alt="Computer"
      data-nimg="1"
      decoding="async"
      height="500"
      loading="lazy"
      src="/_next/image?url=%2Fassert%2Fdemo-product.jpg&w=750&q=75"
      srcset="/_next/image?url=%2Fassert%2Fdemo-product.jpg&w=384&q=75 1x, /_next/image?url=%2Fassert%2Fdemo-product.jpg&w=750&q=75 2x"
      style="color: transparent; background-size: cover; background-position: 50% 50%; background-repeat: no-repeat;"
      width="350"
    />
  </div>
  <div
    class="col-span-12 lg:col-span-7 flex  justify-center items-center"
  >
    <div
      class="space-y-3  text-center lg:text-left"
    >
      <p
        class="text-sm font-semibold"
      >
        Product ID: 
        1
      </p>
      <p
        class="text-xl font-extrabold font-[#7F4D4F]"
      >
        Name: 
        Computer
      </p>
      <p
        class="text-lg font-bold  "
      >
        Price: 
        50000
      </p>
      <p
        class="text-[1rem] font-light  "
      >
        Category: 
        Technology
      </p>
      <p
        class="text-[0.8rem] font-light  "
      >
        Description: 
        Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. 
      </p>
      <div
        class="flex justify-center  lg:justify-start items-center lg:items-start"
      >
        <span
          class="cursor-pointer"
        >
          <span
            class="flex justify-center items-center  "
          >
            <span
              class="text-base md:text-base_2  flex justify-center items-center px-[1rem] py-[0.5rem] rounded-[0.563rem] w-[100%] h-[100%] bg-[#EDEDED] hover:bg-[#7A67EE]  duration-[0.5s] text-[#7A67EE] hover:text-[#EDEDED]   border-[0.063rem] hover:border-[#EDEDED] border-[#7A67EE]   "
            >
              <span>
                Add To Cart
              </span>
            </span>
          </span>
        </span>
      </div>
    </div>
  </div>
</div>
`;
